### PR TITLE
Update Web Beacon scoring

### PR DESCRIPTION
**Pull request by lisaSW**
- Updated Beacon Proxy and Beacon SNI to use new scoring metrics
- Added yaml weights for scoring
- updated command line and html reporting

### CODE REVIEW

**Comment by Zalgo2462** [comment]
There was a merge conflict since the merge commit on master did not match the individual commits from the other feature branch. I've force pushed up a version with the merge conflicts resolved.

**Review by Zalgo2462 — CHANGES_REQUESTED** [review]
In the old version, api.twitter.com in the team's proxy dataset scored near perfect, now it appears to be scored very low. Additionally, there appears to be an issue with the proxy beacons analysis and display.

The changes related to SNI beacons looked good to me.

You will likely need to run `git fetch; git reset --hard origin/update-proxy-sni-scoring` in order to reset your local branch to the version which has been rebased on master.

New version:
image

Old version:
image

File: commands/show-beacons-proxy.go [hunk 1]
```diff
@@ -92,16 +92,13 @@ func showBeaconsProxyHuman(data []beaconproxy.Result, showNetNames bool) error {
 			row = []string{
 				f(d.Score), d.SrcNetworkName,
 				d.SrcIP, d.FQDN, d.Proxy.NetworkName, d.Proxy.IP,
-				i(d.Connections), i(d.Ts.Range), i(d.Ts.Mode),
-				i(d.Ts.ModeCount), f(d.Ts.Skew),
-				i(d.Ts.Dispersion),
+				i(d.Connections), f(d.Ts.Score), f(d.DurScore), f(d.HistScore), i(d.Ts.Mode),
```
Comment (Zalgo2462) [inline commands/show-beacons-proxy.go:95]: It looks like the headers need to be updated in this method.

**Review by Zalgo2462 — COMMENTED** [review]
New api.twitter.com json doc:
```js
{
    "_id" : ObjectId("63a0d4a961d77e7b26e71513"),
    "fqdn" : "api.twitter.com",
    "src" : "192.168.0.15",
    "src_network_uuid" : UUID("ffffffff-ffff-ffff-ffff-fffffffffffe"),
    "bucket_divs" : [ 
        NumberLong(1610645063), 
        NumberLong(1610645622), 
        NumberLong(1610646181), 
        NumberLong(1610646740), 
        NumberLong(1610647299), 
        NumberLong(1610647858), 
        NumberLong(1610648417), 
        NumberLong(1610648976), 
        NumberLong(1610649535), 
        NumberLong(1610650094), 
        NumberLong(1610650653), 
        NumberLong(1610651212), 
        NumberLong(1610651771), 
        NumberLong(1610652330), 
        NumberLong(1610652889), 
        NumberLong(1610653448), 
        NumberLong(1610654007), 
        NumberLong(1610654566), 
        NumberLong(1610655125), 
        NumberLong(1610655684), 
        NumberLong(1610656243), 
        NumberLong(1610656802), 
        NumberLong(1610657361), 
        NumberLong(1610657920), 
        NumberLong(1610658495)
    ],
    "cid" : 0,
    "connection_count" : NumberLong(14303),
    "duration_score" : 0.542,
    "freq_count" : {
        "287" : 1,
        "0" : 19,
        "479" : 1,
        "495" : 1,
        "259" : 1,
        "3" : 1
    },
    "freq_list" : [ 
        287, 
        0, 
        0, 
        0, 
        0, 
        0, 
        0, 
        0, 
        0, 
        0, 
        479, 
        495, 
        259, 
        3, 
        0, 
        0, 
        0, 
        0, 
        0, 
        0, 
        0, 
        0, 
        0, 
        0
    ],
    "hist_score" : 0.0,
    "proxy" : {
        "ip" : "142.93.18.199",
        "network_uuid" : UUID("ffffffff-ffff-ffff-ffff-ffffffffffff"),
        "network_name" : "Public"
    },
    "score" : 0.514,
    "src_network_name" : "Unknown Private",
    "ts" : {
        "conns_score" : 1.0,
        "dispersion" : NumberLong(0),
        "interval_counts" : [ 
            NumberLong(1275), 
            NumberLong(139), 
            NumberLong(64), 
            NumberLong(22), 
            NumberLong(6), 
            NumberLong(1), 
            NumberLong(4), 
            NumberLong(4), 
            NumberLong(3), 
            NumberLong(3), 
            NumberLong(1)
        ],
        "intervals" : [ 
            NumberLong(1), 
            NumberLong(2), 
            NumberLong(3), 
            NumberLong(4), 
            NumberLong(5), 
            NumberLong(6), 
            NumberLong(7), 
            NumberLong(8), 
            NumberLong(9), 
            NumberLong(10), 
            NumberLong(5292)
        ],
        "mode" : NumberLong(1),
        "mode_count" : NumberLong(1275),
        "range" : NumberLong(5291),
        "score" : 1.0,
        "skew" : 0.0
    }
}
```

Old doc:
```js
{
    "_id" : ObjectId("63a0d63861d77e7b26e8899d"),
    "fqdn" : "api.twitter.com",
    "src" : "192.168.0.15",
    "src_network_uuid" : UUID("ffffffff-ffff-ffff-ffff-fffffffffffe"),
    "cid" : 0,
    "connection_count" : NumberLong(14303),
    "proxy" : {
        "ip" : "142.93.18.199",
        "network_uuid" : UUID("ffffffff-ffff-ffff-ffff-ffffffffffff"),
        "network_name" : "Public"
    },
    "score" : 1.0,
    "src_network_name" : "Unknown Private",
    "ts" : {
        "conns_score" : 1.0,
        "dispersion" : NumberLong(0),
        "interval_counts" : [ 
            NumberLong(12780), 
            NumberLong(1275), 
            NumberLong(139), 
            NumberLong(64), 
            NumberLong(22), 
            NumberLong(6), 
            NumberLong(1), 
            NumberLong(4), 
            NumberLong(4), 
            NumberLong(3), 
            NumberLong(3), 
            NumberLong(1)
        ],
        "intervals" : [ 
            NumberLong(0), 
            NumberLong(1), 
            NumberLong(2), 
            NumberLong(3), 
            NumberLong(4), 
            NumberLong(5), 
            NumberLong(6), 
            NumberLong(7), 
            NumberLong(8), 
            NumberLong(9), 
            NumberLong(10), 
            NumberLong(5292)
        ],
        "mode" : NumberLong(0),
        "mode_count" : NumberLong(12780),
        "range" : NumberLong(5291),
        "score" : 1.0,
        "skew" : 0.0
    }
}
```

It looks like this beacon was connecting very rapidly with 0-intervals being the majority of the intervals.

**Comment by Zalgo2462** [comment]
All of the hist scores appear to be zero for the proxy dataset.

**Comment by Zalgo2462** [comment]
The proxy dataset only spans 3.73 hours (13432 seconds). Maybe this is an edge case for the hist score?

**Comment by Zalgo2462** [comment]
Looking at `freq_list`, this beacon was really only active for around an hour. I think it makes sense that it should be scored lower.

**Comment by Zalgo2462** [comment]
TL;DR: The headers for the human CLI output need fixed up for proxy beacons. I believe the other issues I was seeing were being caused by my testing data. Everything else LGTM.

**Review by Zalgo2462 — APPROVED** [review]
In addition to bringing the Web beacon scoring in line with the IP beacon scoring, this PR applies part 1 of #773 to the proxy beacon scoring algorithm as well. 

Everything looks good to me.
